pkg/installmanager/installmanager.go

```diff
@@ -766,6 +773,50 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment) error {
 		}
 	}
 
+	// Day 0: Configure Hive worker MachinePool manifests with additional security group when
+	// ExtraWorkerSecurityGroupAnnotation has been set in annotations for the worker MachinePool.
+	// For details, see HIVE-1802.
+	if metav1.HasAnnotation(workerMachinePool.ObjectMeta, constants.ExtraWorkerSecurityGroupAnnotation) {
+		m.log.Info("modifying worker machineset manifests")
+		// Retrieve list of worker MachinePool manifest files
+		machineSetManifests := []string{}
+		err := filepath.Walk(filepath.Join(m.WorkDir, "openshift"), func(path string, info os.FileInfo, err error) error {
+			if !info.IsDir() && strings.Contains(path, "worker-machineset") {
+				machineSetManifests = append(machineSetManifests, path)
+			}
+			return nil
+		})
+		if err != nil {
+			m.log.WithError(err).Error("error finding machine pool manifests")
+			return err
+		}
+		// Modify yaml to include the additional security group
+		ops := []byte(fmt.Sprintf(`---
+- op: add
+  path: /spec/template/spec/providerSpec/value/securityGroups/0/filters/0/values/1
```

Comment:
As we discussed, let's see if we can append without assuming the size of the existing list via:
```suggestion
  path: /spec/template/spec/providerSpec/value/securityGroups/0/filters/0/values/-
```